code/game/gamemodes/cult/runes.dm

```diff
@@ -452,7 +452,7 @@ structure_check() searches for nearby cultist structures required for the invoca
 
 	var/input_rune_key = input(user, "Choose a rune to teleport to.", "Rune to Teleport to") as null|anything in potential_runes //we know what key they picked
 	var/obj/effect/rune/teleport/actual_selected_rune = potential_runes[input_rune_key] //what rune does that key correspond to?
-	if(!src || !Adjacent(user) || QDELETED(src) || user.incapacitated() || !actual_selected_rune)
+	if(!src || QDELETED(src) || !actual_selected_rune || QDELETED(actual_selected_rune) ||!Adjacent(user) || user.incapacitated())
```

Comment:
split the infrastructural "is this thing qdelled?" and the business logic "is this human capable of the task?" into two checks maybe?

Reply:
It'd definitely be more readable if they were split up, but there would need to be the same thing copy-pasted twice for it to work.
e.g.
```dm
if(QDELETED(stuff))
	fail_invoke()
	return

if(!can_use_or_whatever(user))
	fail_invoke()
	return
```
Unless you can think of a better way of doing this, I think it would be better to just leave it as-is.

Reply:
I think two `fail_invoke()` calls are fine like that.